docs/whats_new.md

```diff
@@ -184,6 +184,41 @@ To enable the routing changes above we had to slightly adjust the signature of t
 +    Add(methods []string, path string, handler Handler, middleware ...Handler) Router
 ```
 
+### Test Config
+
+The `app.Test()` method now allows users to customize their test configurations:
+
+<details>
+<summary>Example</summary>
+
+```go
+// Create a test app with a handler to test
+app := fiber.New()
+app.Get("/", func(c fiber.Ctx) {
+  return c.SendString("hello world")
+})
+
+// Define the HTTP request and custom TestConfig to test the handler
+req := httptest.NewRequest(MethodGet, "/", nil)
+testConfig := fiber.TestConfig{
+  Timeout:       -1,
+  FailOnTimeout: false,
+}
+
+// Test the handler using the request and testConfig
+resp, err := app.Test(req, testConfig)
+```
```

Comment:
This does sound like a much better example to show off, but I think it would be best to add this example use case in this separate PR:

🔥 Feature (v3): Add buffered streaming support #3131 

Before doing so, I think we would need to merge these changes first so we can add unit tests using the new `fiber.TestConfig{}` feature from this PR.

Please let me know your thoughts!